src/components/advises/add-advise-dialog-form.tsx

```diff
@@ -0,0 +1,61 @@
+'use client';
+
+import { createAdvise } from "@/actions/advises/create-advise";
+import { AdviseFormData, adviseSchema } from "@/schemas/advise-schema";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useForm } from "react-hook-form";
+import { toast } from "sonner";
+import { Form, FormControl, FormField, FormItem, FormMessage } from "../ui/form";
+import { Textarea } from "../ui/textarea";
+import { Button } from "../ui/button";
+import { Dispatch, SetStateAction } from "react";
+
+interface DialogFormProps {
+  setDialogOpen(data: boolean): void;
+}
+
+export const AddAdviseDialogForm = (props: DialogFormProps) => {
+  const form = useForm<AdviseFormData>({
+    resolver: zodResolver(adviseSchema),
+    defaultValues: {
+      content: '',
+    },
+  });
+
+  function onSubmit({ content }: AdviseFormData) {
+    toast.promise(createAdvise(content), {
+      loading: 'Creando consejo...',
+      success: () => {
+        form.reset();
+        return 'Tu consejo fue publicado exitosamente. Gracias por compartir!';
+      },
+      error: 'Ocurrió un error al publicar el consejo',
+    });
+
+    props.setDialogOpen(false);
+  }
```

Comment:
Podemos usar el patrón de parameter destructuring para simplificar un toque el código.
```suggestion
export const AddAdviseDialogForm = ({ setDialogOpen }: DialogFormProps) => {
  const form = useForm<AdviseFormData>({
    resolver: zodResolver(adviseSchema),
    defaultValues: {
      content: '',
    },
  });

  function onSubmit({ content }: AdviseFormData) {
    toast.promise(createAdvise(content), {
      loading: 'Creando consejo...',
      success: () => {
        form.reset();
        return 'Tu consejo fue publicado exitosamente. Gracias por compartir!';
      },
      error: 'Ocurrió un error al publicar el consejo',
    });

    setDialogOpen(false);
  }
```